app/tests/spec/views/sign_in.js

```diff
@@ -203,6 +203,27 @@ function (chai, $, sinon, p, View, Session, AuthErrors, OAuthErrors, Metrics,
           });
       });
 
+      it('redirects users to the page they requested before getting redirected to signin', function () {
+        view._redirectTo = 'settings/avatar/change';
```

Comment:
To respect encapsulation we should set this using the ephemeral data API rather than modify the pseudo-private property.